public/app/core/components/ThresholdManager/ThresholdForm.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { ThresholdEditor } from './ThresholdEditor';
+import { react2AngularDirective } from 'app/core/utils/react2angular';
+
+export interface IProps {
+  thresholds: any[];
+  onChange: (threshold: any) => any;
+}
+
+export class ThresholdForm extends React.Component<IProps, any> {
+  constructor(props) {
+    super(props);
+    this.onChange = this.onChange.bind(this);
+    this.onRemove = this.onRemove.bind(this);
+    this.addThreshold = this.addThreshold.bind(this);
+  }
+
+  onChange(threshold, index) {
+    this.props.onChange(this.props.thresholds);
+  }
+
+  onRemove(index) {
+    let newThresholds = this.props.thresholds.filter((t, i) => i !== index);
+    this.props.onChange(newThresholds);
+  }
+
+  addThreshold() {
+    let newThresholds = this.props.thresholds.concat({
+      value: undefined,
+      colorMode: 'critical',
+      op: 'gt',
+      fill: true,
+      line: true,
+      yaxis: 'left',
+    });
+    this.props.onChange(newThresholds);
+  }
+
+  render() {
+    const thresholdItems = this.props.thresholds.map((threshold, i) => (
+      <ThresholdEditor
+        index={i}
+        key={i.toString()}
+        threshold={threshold}
+        onChange={this.onChange}
+        onRemove={this.onRemove}
+        multipleAxes={false}
+      />
+    ));
+
+    return (
+      <div>
+        {thresholdItems}
+        <div className="gf-form-button-row">
+          <button className="btn btn-inverse" onClick={this.addThreshold}>
+            <i className="fa fa-plus" />&nbsp;Add Threshold
+          </button>
+        </div>
+      </div>
+    );
+  }
+}
+
+react2AngularDirective('thresholdForm', ThresholdForm, [
```

Comment:
there is a reason these are outside the react component, for example in https://github.com/grafana/grafana/blob/master/public/app/core/angular_wrappers.ts as it makes jest test faster/easier as they do not need to read in angular

Reply:
Aha, got it. Just found 2 approaches, so didn't know which should be used.